app/models/authenticator/httpd.rb

```diff
@@ -34,8 +34,8 @@ def failure_reason(_username, request)
       request.headers['X-EXTERNAL-AUTH-ERROR']
     end
 
-    def find_external_identity(_username, user_attrs, membership_list)
-      [user_attrs, membership_list]
+    def find_external_identity(_username, *args)
+      args # should be [user_attributes, membership_list]
```

Comment:
Using the comment to reflect what the code had seems contrary to Ruby's self documenting aspects.

Maybe consider this, which also seems a bit awkward to me but is self documenting.

```ruby
      def find_external_identity(_username, *args)
        user_attrs = args[0]
        membership_list = args[1]
        [user_attrs, membership_list]
      end
```

Reply:
@jvlcek I understand, but it's less efficient, and it would probably get rejected for that reason.

Reply:
I can see that point @djberg96

Reply:
What are the args for the other authenticators...are they user_attrs and membership_list?  I guess I'm more asking if we should change all of the other authenticators instead and be more explicit there.

Reply:
**Currently:**

https://github.com/ManageIQ/manageiq/blob/master/app/models/authenticator/ldap.rb#L74
Defines:
```ruby
def find_external_identity(username, *_args)
```
and

https://github.com/ManageIQ/manageiq/blob/master/app/models/authenticator/httpd.rb#L37
Defines:
```ruby
 def find_external_identity(_username, user_attrs, membership_list)
```

and find_external_identity is only invoke here:

https://github.com/ManageIQ/manageiq/blob/master/app/models/authenticator/base.rb#L112
```bash
identity = find_external_identity(username, *args)
```

**Let's consider this:**

**1** Changing the invocation in base.rb

To:
```bash
  identity = find_external_identity(username, args[0], args[1])
```
**2** Update the function signature in ldap.rb to:

```ruby
      def find_external_identity(username, _user_attrs, _membership_list)
```

**3** Leave the function signature in httpd.rb as:

```ruby
 def find_external_identity(_username, user_attrs, membership_list)
```

If this approach is taken, just to be double safe, I'd like do a couple of end to end tests, which I'd be glad to help with.

Thoughts?

Reply:
I think if `user_attrs` and `membership_list` make sense for the other classes, then we should go with that since it's more stringent. If you agree @jvlcek and @Fryguy I'll make the changes as part of this PR.

Reply:
@djberg96 👍 to the approach I outlined above -> https://github.com/ManageIQ/manageiq/pull/19533#discussion_r348775745
Why don't you update the files to reflect that approach and we can move forward from there.